### PR TITLE
[RISC-V] Fix Microsoft.VisualBasic.Tests.ConversionTests

### PR DESCRIPTION
**Pull request by yurai007**
In order to fix Microsoft.VisualBasic.Tests.ConversionTests we need to:

- replace NYI_RISCV64s with proper float/double registers initialization in genZeroInitFltRegs
- fix jump distance binding logic by performing INS_OPTS_J_cond jump transformation when jmpDist exceed the maximum short distance

After this change Microsoft.VisualBasic.Tests.ConversionTests starts passing without triggering any CoreCLR assertion.

Part of https://github.com/dotnet/runtime/issues/84834
cc @wscho77 @HJLeee @clamp03 @JongHeonChoi @t-mustafin @gbalykov @ashaurtaev @sirntar @tomeksowi

### CODE REVIEW

**Comment by yurai007** [comment]
@dotnet-policy-service agree

**Comment by clamp03** [comment]
@yurai007 Please update code format by executing jit-format (https://github.com/dotnet/jitutils/blob/main/doc/formatting.md) or checking CI error message.

**Comment by yurai007** [comment]
Fix formatting

**Review by tomeksowi — APPROVED** [review]

**Comment by yurai007** [comment]
@jakobbotsch Ping. Could you please take a look?

File: src/coreclr/jit/emitriscv64.cpp [hunk 1]
```diff
@@ -1666,11 +1666,41 @@ void emitter::emitJumpDistBind()
...
         emitDispIGlist(true);
     }
 #endif
 
+#if DEBUG_EMIT
+    auto printJmpInfo = [this](const instrDescJmp* jmp, const insGroup* jmpIG, NATIVE_OFFSET extra,
+                               UNATIVE_OFFSET srcInstrOffs, UNATIVE_OFFSET srcEncodingOffs, UNATIVE_OFFSET dstOffs,
+                               NATIVE_OFFSET jmpDist, const char* direction) {
+        assert(jmp->idDebugOnlyInfo() != nullptr);
+        const char* dirId = (strcmp(direction, "fwd") == 0) ? "[1]" : "[2]";
```
Comment (jakobbotsch) [inline src/coreclr/jit/emitriscv64.cpp:1679]: Move this into the then block below? This code is going to run in all debug/checked builds.

**Review by jakobbotsch — APPROVED** [review]

**Comment by clamp03** [comment]
@jakobbotsch Can this be merged?

**Comment by jakobbotsch** [comment]
> @jakobbotsch Can this be merged?

Done. Sorry, I was waiting for CI to finish when I approved.

**Comment by clamp03** [comment]
@jakobbotsch Thank you so much!!!
@yurai007 Congratulations on your fist commit! 🎉🎉🎉

**Pull request merged**